src/cloudflare/internal/pipeline-transform.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import entrypoints from 'cloudflare-internal:workers'
+
+async function* readLines(stream: ReadableStream<string>): AsyncGenerator<string> {
+  let partial = ''
+
+  // @ts-ignore
+  for await (const chunk of stream) {
+    const full = partial + chunk
+    const lines = full.split('\n')
```

Comment:
Probably a later optimization, but could probably implement a streaming split to avoid extra memory allocations.  Might not be worth the effort, if the chunks are relatively small.